src/lib/y2partitioner/confirm_recursive_delete.rb

```diff
@@ -0,0 +1,112 @@
+# encoding: utf-8
+
+# Copyright (c) [2018] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+
+module Y2Partitioner
+  # Mixin for confimation when recursive deletion is performed
+  module ConfirmRecursiveDelete
+  private
+
+    # Helpful method to show a descriptive confirm message with all affected devices
+    #
+    # @param devices [Array<Y2Storage::Device>, Y2Storage::Device] devices to delete
+    # @param headline [String]
+    # @param label_before [String]
+    # @param label_after [String]
+    # @return [Boolean]
+    def confirm_recursive_delete(devices, headline, label_before, label_after)
+      button_box = ButtonBox(
+        PushButton(Id(:yes), Opt(:okButton), Yast::Label.DeleteButton),
+        PushButton(
+          Id(:no_button),
+          Opt(:default, :cancelButton),
+          Yast::Label.CancelButton
+        )
+      )
+
+      fancy_question(headline,
+        label_before,
+        Yast::HTML.List(devices_to_delete(devices).sort),
+        label_after,
+        button_box)
+    end
+
+    # @param headline [String]
+    # @param label_before [String]
+    # @param rich_text [String]
+    # @param label_after [String]
+    # @param button_term [Yast::UI::Term]
+    # @return [Boolean]
+    def fancy_question(headline, label_before, rich_text, label_after, button_term)
```

Comment:
maybe this label before and label after can be added to Yast2::Popup so we can use common method.

Reply:
Yes, that would be nice.

Reply:
Is there any way to do exactly the same with current Yast2::Popup ?

Reply:
I worry not, you can have headline + richtext, but not before/after label. We already face it before with some other dialog, so I should add it to my todo list.